web/docs/auth/overview.md

```diff
@@ -637,3 +1013,238 @@ The default value is `"/"`.
 :::note
 Automatic redirect on successful login only works when using the Wasp-provided [Auth UI](/docs/auth/ui).
 :::
+
+#### `signup: SignupOptions`
+
+Read more about the signup process customization API in the [Signup Fields Customization](#signup-fields-customization) section.
+
+
+### Signup Fields Customization
+
+If you want to add extra fields to the signup process, the backend needs to know how to save them to the database. You do that by defining the `auth.signup.additionalFields` field in your `main.wasp` file.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp" {9-11}
+app crudTesting {
+  // ...
+  auth: {
+    userEntity: User,
+    methods: {
+      usernameAndPassword: {},
+    },
+    onAuthFailedRedirectTo: "/login",
+    signup: {
+      additionalFields: import { fields } from "@server/auth/signup.js",
+    },
+  },
+}
+```
+
+Then we'll export the `fields` object from the `server/auth/signup.js` file:
+
+```ts title="server/auth/signup.js"
+import { defineAdditionalSignupFields } from "@wasp/auth/index.js";
+
+export const fields = defineAdditionalSignupFields({
+  address: async (data) => {
+    const address = data.address;
+    if (typeof address !== "string") {
+      throw new Error("Address is required");
+    }
+    if (address.length < 5) {
+      throw new Error("Address must be at least 5 characters long");
+    }
+    return address;
+  },
+});
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp" {9-11}
+app crudTesting {
+  // ...
+  auth: {
+    userEntity: User,
+    methods: {
+      usernameAndPassword: {},
+    },
+    onAuthFailedRedirectTo: "/login",
+    signup: {
+      additionalFields: import { fields } from "@server/auth/signup.js",
+    },
+  },
+}
+```
+
+Then we'll export the `fields` object from the `server/auth/signup.ts` file:
+
+```ts title="server/auth/signup.ts"
+import { defineAdditionalSignupFields } from "@wasp/auth/index.js";
+
+export const fields = defineAdditionalSignupFields({
+  address: async (data) => {
+    const address = data.address;
+    if (typeof address !== "string") {
+      throw new Error("Address is required");
+    }
+    if (address.length < 5) {
+      throw new Error("Address must be at least 5 characters long");
+    }
+    return address;
+  },
+});
+```
+
+</TabItem>
+</Tabs>
+
+The `fields` object is an object where the keys represent the field name, and the values are functions which receives the data sent from the client and return the value of the field.
+
+If the field value is invalid, the function should throw an error.
+
+The field details object has the following properties:
+
```

Comment:
This stuff is same as above -> is there any way we could avoid this duplication?

Reply:
If we want to keep the API reference independent, there will be some duplication always.

Reply:
But this is identical, right? The more duplication we have, more chances of it going out of sync. I know we can't remove all duplication, some of it is semantical, but this seems like literal copy paste, that is why I am asking, also for the future, if not now.
We could maybe, when we have duplication like this, extract it into a React component?

Reply:
> some of it is semantical, but this seems like literal copy paste

Once we pull out this content into a API Reference component, then it needs to be standalone. I could pull out this piece of repetition into a tiny React component. But then we need to make sure that tiny piece fits in both contexts over time (for example, it can't refer to something above or below itself - since it's used in two difference places) which think is worse than having some duplication.

Reply:
Sure, if it can't be the same in both places, that is fine, but in case we do have pieces that are completely the same, I think it is worth it investigating extracting them.